### PR TITLE
[CU-2dve0nb] docs(Technical stack overview): Replaced old page with new design

### PR DESCRIPTION
**Pull request by Martin-Kiss**
## Issue
https://app.clickup.com/t/2dve0nb

## Checklist

~~- [ ] I have updated the cargo docs to reflect changes made by this PR _(if applicable)_~~
~~- [ ] I have updated the `book/` to reflect changes made by this PR _(if applicable)_~~

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2dve0nb mdbook changes](https://app.clickup.com/t/2dve0nb)

File: book/src/products/technical-stack-overview.md
Comment (cor) [inline book/src/products/technical-stack-overview.md:7]: why did you make these headings links?

File: book/src/products/technical-stack-overview.md
Comment (cor) [inline book/src/products/technical-stack-overview.md:16]: You should not make `##`'s bold with `**`, as that is done automatically

File: book/src/products/technical-stack-overview.md
Comment (cor) [inline book/src/products/technical-stack-overview.md:23]: You should not make `###`'s bold with `**`, as that is done automatically

**Review by cor — DISMISSED** [review]
Looks good, but needs some changes

File: book/src/products/technical-stack-overview.md
Comment (PoisonPhang) [inline book/src/products/technical-stack-overview.md:26]: This should be a paragraph (no whole line markdown modifiers) not a heading 3

File: book/src/products/technical-stack-overview.md
Comment (PoisonPhang) [inline book/src/products/technical-stack-overview.md:32]: Should be paragraph, not heading 3

File: book/src/products/technical-stack-overview.md
Comment (PoisonPhang) [inline book/src/products/technical-stack-overview.md:38]: Should be paragraph, not heading 3

**Review by PoisonPhang — CHANGES_REQUESTED** [review]

**Review by PoisonPhang — DISMISSED** [review]
LGTM! To fix CI, you might need to rebase on main. 
Feel free to message me on Slack if you'd like any advice on rebasing or need a new approval afterward.

**Review by 0xBrainjar2 — COMMENTED** [review]
Please use new tech stack image jafar has

**Review by PoisonPhang — APPROVED** [review]

**Review by PoisonPhang — APPROVED** [review]

**Comment by KaiserKarel** [comment]
@Martin-Kiss please rebase

**Comment by Martin-Kiss** [comment]
This PR can be closed once #1180 has been merged.

**Comment by Martin-Kiss** [comment]
Redundant PR, closed.

**Pull request closed without merging**
